pkg/cmd/server/server_test.go

```diff
@@ -69,3 +70,87 @@ func TestVeleroResourcesExist(t *testing.T) {
 	veleroAPIResourceList.APIResources = veleroAPIResourceList.APIResources[:3]
 	assert.Error(t, server.veleroResourcesExist())
 }
+
+func TestRemoveControllers(t *testing.T) {
+	logger := velerotest.NewLogger()
+
+	tests := []struct {
+		name                string
+		disabledControllers []string
+		errorExpected       bool
+	}{
+		{
+			name: "Remove one disabable controller",
+			disabledControllers: []string{
+				controller.Backup,
+			},
+			errorExpected: false,
+		},
+		{
+			name: "Remove all disabable controllers",
+			disabledControllers: []string{
+				controller.Backup,
+				controller.BackupDeletion,
+				controller.BackupSync,
+				controller.DownloadRequest,
+				controller.GarbageCollection,
+				controller.ResticRepo,
+				controller.Restore,
+				controller.Schedule,
+				controller.ServerStatusRequest,
+			},
+			errorExpected: false,
+		},
+		{
+			name: "Remove with a non-disabable controller included",
+			disabledControllers: []string{
+				controller.Backup,
+				controller.BackupStorageLocation,
+			},
+			errorExpected: true,
+		},
+		{
+			name: "Remove with a misspelled controller name",
+			disabledControllers: []string{
+				"bakup",
```

Comment:
I'm not sure how to best handle this spelling issue. Would this test still work to have a correctly spelled word, but one that isn't a valid controller name?

Reply:
If by invalid controller name you mean a controller that is not on the list of ones that are disabable, then yes, this test would still work the same. You can see it in this other test: https://github.com/vmware-tanzu/velero/pull/3180/files#diff-073f5a2184977287c11f4b4c94aced2fab5718ee4946a2b050d472fa6059496fR105. Included in the list of controllers to disable is the BSL controller, so there'll be an error.

Is this what you mean?

Reply:
Yeah, I was just thinking of how to trigger the behaviour that we're testing for but fix the codespell issue without adding the word to the list of skipped words as it's something that we could easily misspell elsewhere :)

Reply:
Oh duh, I JUST realized what you mean haha. I like this test because it stands for both "a controller that doesn't exist" and "a controller with a very similar name exists but the system handles it properly". Not married to it though. The check is not a merge blocker. I feel like we should have some sorta rule of thumb for things like this.

Reply:
Yes, we don't want to add this to the list of skipped words. Let me misspell something else.